bin/bb/cmd/utils.go

```diff
@@ -0,0 +1,85 @@
+package cmd
+
+import (
+	"fmt"
+	"regexp"
+	"strings"
+)
+
+// driver://[username[:password]@]host[:port]/[database][?param1=value1&...&paramN=valueN]
+var dsnPattern *regexp.Regexp
+
+func init() {
+	dsnPattern = regexp.MustCompile(
+		`^(?P<driver>.+?)(?::\/\/)` + // driver://
+			`(?:(?P<username>.+?)(?::(?P<password>.+?))?@)?` + // [username[:password]@]
+			`(?P<host>.+?)(?::(?P<port>.+?))?` + // host[:port]
+			`\/(?P<database>.+?)` + // /[database]
+			`(?:\?(?P<params>[^\?]*))?$`) // [?param1=value1&...&paramN=valueN]
+}
+
+type dataSource struct {
+	driver   string
+	username string
+	password string
+	host     string
+	port     string
+	database string
+	params   map[string]string
+
+	sslCA   string // server-ca.pem
+	sslCert string // client-cert.pem
+	sslKey  string // client-key.pem
+}
+
+func parseDSN(dsn string) (*dataSource, error) {
```

Comment:
We may consider using other standard parser instead of implementing our own if possible.

Reply:
See https://sourcegraph.com/github.com/bytebase/bytebase/-/blob/bin/server/cmd/root.go?L359&subtree=true